src/Compilers/Core/Portable/DiagnosticAnalyzer/DefaultAnalyzerAssemblyLoader.Core.cs

```diff
@@ -17,95 +17,48 @@ namespace Microsoft.CodeAnalysis
-        /// </summary>
-        internal static readonly ImmutableHashSet<string> CompilerAssemblySimpleNames =
-           ImmutableHashSet.Create(
-               StringComparer.OrdinalIgnoreCase,
-               "Microsoft.CodeAnalysis",
-               "Microsoft.CodeAnalysis.CSharp",
-               "Microsoft.CodeAnalysis.VisualBasic",
-               "System.Collections",
-               "System.Collections.Concurrent",
-               "System.Collections.Immutable",
-               "System.Console",
-               "System.Diagnostics.Debug",
-               "System.Diagnostics.StackTrace",
-               "System.Diagnostics.Tracing",
-               "System.IO.Compression",
-               "System.IO.FileSystem",
-               "System.Linq",
-               "System.Linq.Expressions",
-               "System.Memory",
-               "System.Reflection.Metadata",
-               "System.Reflection.Primitives",
-               "System.Resources.ResourceManager",
-               "System.Runtime",
-               "System.Runtime.CompilerServices.Unsafe",
-               "System.Runtime.Extensions",
-               "System.Runtime.InteropServices",
-               "System.Runtime.InteropServices.RuntimeInformation",
-               "System.Runtime.Loader",
-               "System.Runtime.Numerics",
-               "System.Runtime.Serialization.Primitives",
-               "System.Security.Cryptography.Algorithms",
-               "System.Security.Cryptography.Primitives",
-               "System.Text.Encoding.CodePages",
-               "System.Text.Encoding.Extensions",
-               "System.Text.RegularExpressions",
-               "System.Threading",
-               "System.Threading.Tasks",
-               "System.Threading.Tasks.Parallel",
-               "System.Threading.Thread",
-               "System.Threading.ThreadPool",
-               "System.Xml.ReaderWriter",
-               "System.Xml.XDocument",
-               "System.Xml.XPath.XDocument");
-
-        internal virtual ImmutableHashSet<string> AssemblySimpleNamesToBeLoadedInCompilerContext => CompilerAssemblySimpleNames;
-
-        // This is the context where compiler (and some of its dependencies) are being loaded into, which might be different from AssemblyLoadContext.Default.
-        private static readonly AssemblyLoadContext s_compilerLoadContext = AssemblyLoadContext.GetLoadContext(typeof(DefaultAnalyzerAssemblyLoader).GetTypeInfo().Assembly)!;
-
+        private readonly AssemblyLoadContext _compilerLoadContext;
         private readonly object _guard = new object();
         private readonly Dictionary<string, DirectoryLoadContext> _loadContextByDirectory = new Dictionary<string, DirectoryLoadContext>(StringComparer.Ordinal);
 
-        protected override Assembly LoadFromPathUncheckedImpl(string fullPath)
+        internal AssemblyLoadContext CompilerLoadContext => _compilerLoadContext;
+
+        internal DefaultAnalyzerAssemblyLoader()
```

Comment:
It feels like it would be more hygienic to either remove the parameter default value in the subsequent constructor, or to delete this constructor.

Reply:
removed the empty ctor